cpp/src/barretenberg/honk/pcs/commitment_key.hpp

```diff
@@ -65,6 +66,26 @@ class CommitmentKey {
             const_cast<Fr*>(polynomial.data()), srs.get_monomial_points(), degree, pippenger_runtime_state);
     };
 
+    /**
+     * @brief Add computation of commitment C = [p(x)] = ∑ᵢ aᵢ⋅[xⁱ]₁ to the provided work queue
+     *
+     * @param polynomial a univariate polynomial p(X) = ∑ᵢ aᵢ⋅Xⁱ ()
+     */
+    void queue_commitment(Polynomial& polynomial, std::string& commitment_tag, bonk::work_queue& work_queue)
```

Comment:
Wondering if IPA needs similar kind of treatment both in the commit function and inside `reduce_prove` and `reduce_verify` in `ipa.hpp` for MSMs. Maybe we can add them later? Would like to know your opinion on this.

Reply:
Also I feel like we should give a brief explanation of what is `work_queue` and why we need them in the PR description.